helix-core/src/main/java/org/apache/helix/task/WorkflowConfig.java

```diff
@@ -322,6 +322,22 @@ public static WorkflowConfig fromHelixProperty(HelixProperty property)
     return Builder.fromMap(configs).setWorkflowId(property.getId()).build();
   }
 
+  /**
+   * Build a WorkflowConfig from a HelixProperty.
+   *
+   * @param property a HelixProperty
+   * @return a WorkflowConfig if the property is a valid WorkflowConfig.
+   *         Otherwise return null
+   */
+  public static WorkflowConfig parseWorkflowConfig(HelixProperty property)
+      throws IllegalArgumentException {
+    Map<String, String> configs = property.getRecord().getSimpleFields();
+    if (!configs.containsKey(WorkflowConfigProperty.Dag.name())) {
```

Comment:
Validation logic is better to be located within WorkflowConfig class.
Here we just trigger the check and then convert.

Reply:
Yes. The reason I put it here was because I didn't want to change the existed code as being told. And I would like to re-use the var `configs` in the following code: `return Builder.fromMap(configs).setWorkflowId(property.getId()).build();`

FYI, how do you like to use Optional as we discussed about in the above conversations?

Reply:
The previous discussion was not based on the code so hard for me to track. Might be easier if Hunter is back and we can talk. But if that is just for the timeout, I think max integer or max long would be good enough.

Reply:
My preference would be to use number 0 to represent timeout disabled. I see quite many timeout settings are designed like this, ex, in Hadoop.